azure_functions_worker/dispatcher.py

```diff
@@ -62,24 +63,20 @@ def __init__(self, loop: BaseEventLoop, host: str, port: int,
 
         self._old_task_factory = None
 
-        # A thread-pool for synchronous function calls.  We limit
-        # the number of threads to 1 so that one Python worker can
-        # only run one synchronous function in parallel.  This is
-        # because synchronous code in Python is rarely designed with
-        # concurrency in mind, so we don't want to allow users to
-        # have races in their synchronous functions.  Moreover,
-        # because of the GIL in CPython, it rarely makes sense to
-        # use threads (unless the code is IO bound, but we have
-        # async support for that.)
-        self._sync_call_tp = concurrent.futures.ThreadPoolExecutor(
-            max_workers=1)
-
-        self._grpc_connect_timeout = grpc_connect_timeout
+        # We allow the customer to change synchronous thread pool count by
+        # PYTHON_THREADPOOL_THREAD_COUNT app setting. The default value is 1.
+        self._sync_tp_max_workers: int = int(get_app_setting(
+            PYTHON_THREADPOOL_THREAD_COUNT, '1'))
```

Comment:
If the customer puts in an incorrect value, it might throw an exception when casting to int - you will need to take care of that - try-except, etc.

Reply:
Also - I feel that we should keep a true max - like 32 or 64? We don't want customers to put in 10000 in there (I understand it is thus a theoretical value and python only creates a new thread when it needs it and not immediately)

Reply:
I will cap the max value to 32, check the 3.8 implementation below https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor